pkg/tfshim/sdk-v2/provider.go

```diff
@@ -62,9 +63,16 @@ func NewProvider(p *schema.Provider, opts ...providerOption) shim.Provider {
 		tf:   p,
 		opts: opts,
 	}
-	if cmdutil.IsTruthy(os.Getenv("PULUMI_ENABLE_PLAN_RESOURCE_CHANGE")) {
+
+	envPRC := os.Getenv("PULUMI_ENABLE_PLAN_RESOURCE_CHANGE")
+	if cmdutil.IsTruthy(envPRC) {
 		return newProviderWithPlanResourceChange(p, prov, func(s string) bool { return true })
 	}
+
+	if strings.EqualFold(envPRC, "false") || envPRC == "0" {
+		return prov
+	}
```

Comment:
Lets add a property trinary here and warn if we get something else. I don't want users to set `PULUMI_ENABLE_PLAN_RESOURCE_CHANGE=disable` and have it silently do nothing.
```suggestion
	const envPRCVar = "PULUMI_ENABLE_PLAN_RESOURCE_CHANGE"
	envPRC := os.Getenv(envPRCVar)
	switch {
	case cmdutil.IsTruthy(envPRC):
		return newProviderWithPlanResourceChange(p, prov, func(s string) bool { return true })
	case strings.EqualFold(envPRC, "false") || envPRC == "0":
		return prov
	case "":
		// Do nothing, we will apply the default below.
	default:
		// The user set PULUMI_ENABLE_PLAN_RESOURCE_CHANGE to a value that isn't valid.
		warn("%s is not a valid value for %s", envPRC, envPRCVar)
	}
```